linkerd/app/outbound/src/opaq/logical/route.rs

```diff
@@ -77,13 +96,21 @@ where
                 // consideration, so we must eagerly fail requests to prevent
                 // leaking tasks onto the runtime.
                 .push_on_service(svc::LoadShed::layer())
+                .push_filter(|rt: Self| -> Result<_, Error> {
+                    if rt.params.forbidden {
+                        Err(Errno::from(Code::ECONNREFUSED).into())
+                    } else {
+                        Ok(rt)
+                    }
+                })
```

Comment:
I don't think we want to use Errno -- the existing IO errors won't be represented like this.

In a first version, we should probably opt for bespoke (thiserror) types indicating Forbidden (and Internal?) errors. We may need to adjust that in light of metrics but for now it seems like a starting point that will at least show up in the logs.

I like using push_filter for this, though 👍 .